lib/activity.php

```diff
@@ -0,0 +1,222 @@
+	 */
+	public function getTypeIcon($type) {
+		switch ($type) {
+			case self::TYPE_VIRUS_DETECTED:
+				return 'icon-info';
+		}
+		return false;
+	}
+
+	/**
+	 * The extension can translate a given message to the requested languages.
+	 * If no translation is available false is to be returned.
+	 *
+	 * @param string $app
+	 * @param string $text
+	 * @param array $params
+	 * @param boolean $stripPath
+	 * @param boolean $highlightParams
+	 * @param string $languageCode
+	 * @return string|false
+	 */
+	public function translate($app, $text, $params, $stripPath, $highlightParams, $languageCode) {
+		$l = $this->getL10N($languageCode);
+
+		if ($app === self::APP_NAME) {
+			switch ($text) {
+				case self::SUBJECT_VIRUS_DETECTED:
+					return (string) $l->t('File %s is infected with %s', $params);
+				case self::MESSAGE_FILE_DELETED:
+					return (string) $l->t('It is going to be deleted', $params);
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * The extension can define the type of parameters for translation
+	 *
+	 * Currently known types are:
+	 * * file		=> will strip away the path of the file and add a tooltip with it
+	 * * username	=> will add the avatar of the user
+	 *
+	 * @param string $app
+	 * @param string $text
+	 * @return array|false
+	 */
+	public function getSpecialParameterList($app, $text) {
+		return false;
+	}
+
+	/**
+	 * The extension can define the parameter grouping by returning the index as integer.
+	 * In case no grouping is required false is to be returned.
+	 *
+	 * @param array $activity
+	 * @return integer|false
+	 */
+	public function getGroupParameter($activity) {
+		return false;
+	}
+
+	/**
+	 * The extension can define additional navigation entries. The array returned has to contain two keys 'top'
+	 * and 'apps' which hold arrays with the relevant entries.
+	 * If no further entries are to be added false is no be returned.
+	 *
+	 * @return array|false
+	 */
+	public function getNavigation() {
+		$l = $this->getL10N();
+		return [
+			'apps' => [],
+			'top' => [
+				self::FILTER_AVIR => [
+					'id' => self::FILTER_AVIR,
+					'name' => (string) $l->t('Antivirus'),
+					'url' => $this->URLGenerator->linkToRoute('activity.Activities.showList', ['filter' => self::FILTER_AVIR]),
+				],
+			],
+		];
+	}
+
+	/**
+	 * The extension can check if a customer filter (given by a query string like filter=abc) is valid or not.
+	 *
+	 * @param string $filterValue
+	 * @return boolean
+	 */
+	public function isFilterValid($filterValue) {
+		return $filterValue === self::FILTER_AVIR;
+	}
+
+	/**
+	 * The extension can filter the types based on the filter if required.
+	 * In case no filter is to be applied false is to be returned unchanged.
+	 *
+	 * @param array $types
+	 * @param string $filter
+	 * @return array|false
+	 */
+	public function filterNotificationTypes($types, $filter) {
+		switch ($filter) {
+			case self::FILTER_AVIR:
+				return array_intersect([self::TYPE_VIRUS_DETECTED], $types);
```

Comment:
I'd just do 
```php
return [self::TYPE_VIRUS_DETECTED];
```
here. This avoids an empty list, when the user hides them from his default stream